scipy/linalg/_basic.py

```diff
@@ -59,19 +81,20 @@ def solve(a, b, lower=False, overwrite_a=False,
     corresponding string to ``assume_a`` key chooses the dedicated solver.
     The available options are
 
-    ===================  ========
-     generic matrix       'gen'
-     symmetric            'sym'
-     hermitian            'her'
-     positive definite    'pos'
-    ===================  ========
-
-    If omitted, ``'gen'`` is the default structure.
-
-    The datatype of the arrays define which solver is called regardless
-    of the values. In other words, even when the complex array entries have
-    precisely zero imaginary parts, the complex solver will be called based
-    on the data type of the array.
+    =============================  =======
+     diagonal                       'dia'
+     tri-diagonal                   'tri'
```

Comment:
`tri` is probably better reserved for triangular. The users will solve triangular compared to tridiagonal is probably order of magnitude more. We can use more letters if need be with `triu` or `tril`, and then `trid` for tridiagonal. We don't need to stick to three letters, in fact we can spell them out completely. Don't have any strong preference but ergonomics-wise `tri` should somehow map to triangular.

Reply:
I had `triu` and `tril` originally, but then I tried to stick to three letters. I'm fine with spelling them out.